### PR TITLE
en/config/dns.md

### PR DESCRIPTION
**Pull request by rurirei**
i'd interested in dns server and read this page often. sorry for using translation to read and file this. could you review this please

### CODE REVIEW

File: docs/en_US/config/dns.md
Comment (DuckSoft) [inline docs/en_US/config/dns.md:68]:
```suggestion
When its value is a DNS IP address, such as `"8.8.8.8"`, V2Ray will use port 53 of this address for DNS query.
```

File: docs/en_US/config/dns.md
Comment (DuckSoft) [inline docs/en_US/config/dns.md]:
```suggestion
> `Address`: address
```

File: docs/en_US/config/dns.md
Comment (DuckSoft) [inline docs/en_US/config/dns.md]:
```suggestion
 > `ExpectIPs`:\[string\]
```

File: docs/en_US/config/dns.md
Comment (DuckSoft) [inline docs/en_US/config/dns.md]:
```suggestion
Due to the complexity of the DNS protocol, V2Ray only supports the most basic IP queries (A and AAAA record). It is recommended to use local DNS with an additional DNS server for DNS query to fully-feature your DNS service, like [CoreDNS](https://coredns.io/).
```

File: docs/en_US/config/dns.md
Comment (DuckSoft) [inline docs/en_US/config/dns.md]:
```suggestion
Diagram of the server of DNS Procedure is as follows:
```

File: docs/en_US/config/dns.md
Comment (DuckSoft) [inline docs/en_US/config/dns.md]:
```suggestion
 * When its value is a DNS IP address, such as `"8.8.8.8"`, V2Ray will use port 53 of this address for DNS query.
```

File: docs/en_US/config/dns.md
Comment (DuckSoft) [inline docs/en_US/config/dns.md]:
```suggestion
 * When the value is `"localhost"`, it means the machine's preset DNS configuration is used.
```

File: docs/en_US/config/dns.md
Comment (DuckSoft) [inline docs/en_US/config/dns.md]:
```suggestion
 * When the value is in the form of `"https://host:port/dns-query"`, such as `"https://dns.google/dns-query"`, V2Ray will use `DNS over HTTPS` (RFC8484, short for  DOH) to query.  Some service providers have certificates for IP aliases and can write IP directly, such as `https://1.1.1.1/dns-query`.  You can also use non-standard ports and paths, such as `"https://a.b.c.d:8443/my-dns-query"` (4.22.0+)
```

File: docs/en_US/config/dns.md
Comment (DuckSoft) [inline docs/en_US/config/dns.md]:
```suggestion
 * When the value is in the form of `"https+local://host:port/dns-query"`, such as `"https+local://dns.google/dns-query"`, V2Ray will use `DOH local mode`  For query, that is, DOH requests will not go through Routing/Outbound and other components, but directly request externally to reduce time-consuming.  Generally suitable for use on the server side.  Non-standard ports and paths can also be used. (4.22.0+)
```

File: docs/en_US/config/dns.md
Comment (DuckSoft) [inline docs/en_US/config/dns.md]:
```suggestion
 * Plain string: When this domain name completely matches the target domain name, the rule takes effect.  For example, "v2ray.com" matches "v2ray.com" but not "www.v2ray.com".
```

File: docs/en_US/config/dns.md
Comment (DuckSoft) [inline docs/en_US/config/dns.md]:
```suggestion
The DNS clients initialized by different rules will be reflected in the V2Ray startup log with the ʻinfo` level, such as `local DOH`, `remote DOH` and ʻudp` etc. modes.  (4.22.0+)
```

File: docs/en_US/config/dns.md
Comment (KevinZonda) [inline docs/en_US/config/dns.md]:
```suggestion
(V2Ray 4.22.0+) An IP range list, the format is the same as in [Routing Configuration](routing.md#ruleobject).
```
You need to delete all the useless beginning space

**Review by DuckSoft — APPROVED** [review]
The text quality is too high to be accepted. 
However, something is better than nothing. I appreciate your hard work.

Let's merge this, if things are going well.

**Review by DuckSoft — CHANGES_REQUESTED** [review]
Umm... Your hard working wowed me.
However, your literal skills really need to be improved.

Let's settle here and wait for others to refine this.

File: docs/en_US/config/dns.md
Comment (KevinZonda) [inline docs/en_US/config/dns.md]:
```suggestion
`DnsObject` pairs to the item `dns` in the configuration.
```

File: docs/en_US/config/dns.md
Comment (KevinZonda) [inline docs/en_US/config/dns.md]:
```suggestion
 * When the value is in the form of `"https://host:port/dns-query"`, such as `"https://dns.google/dns-query"`, V2Ray will use `DNS over HTTPS` (RFC8484, short for DOH) to query. Some service providers have certificates for IP aliases and can write IP directly, such as `https://1.1.1.1/dns-query`. You can also use non-standard ports and paths, such as `"https://a.b.c.d:8443/my-dns-query"` (4.22.0+)
```

File: docs/en_US/config/dns.md
Comment (KevinZonda) [inline docs/en_US/config/dns.md:74]:
```suggestion
 * When the value is in the form of `"https+local://host:port/dns-query"`, such as `"https+local://dns.google/dns-query"`, V2Ray will use `DOH local mode` For query, that is, DOH requests will not go through Routing/Outbound and other components, but directly request externally to reduce time-consuming. Generally suitable for use on the server side. Non-standard ports and paths can also be used. (4.22.0+)
```

File: docs/en_US/config/dns.md
Comment (KevinZonda) [inline docs/en_US/config/dns.md:84]:
```suggestion
The IP address of the current system is used to inform the server of the location of the client during DNS query. It cannot be a private address.
```

File: docs/en_US/config/dns.md
Comment (KevinZonda) [inline docs/en_US/config/dns.md]:
```suggestion
(V2Ray 4.13+) The query traffic sent by this DNS, except for `localhost` and `DOHL_` modes, will carry this identifier, which can be matched with `inboundTag` in the routing.
```

File: docs/en_US/config/dns.md
Comment (KevinZonda) [inline docs/en_US/config/dns.md:111]:
```suggestion
DNS server port, such as `53`. This item defaults to `53` by default. When using DOH mode, this item is invalid, non-standard port should be specified in URL.
```

File: docs/en_US/config/dns.md
Comment (KevinZonda) [inline docs/en_US/config/dns.md]:
```suggestion
A list of domain names. The domain names contained in this list will be queried by this server first. The domain name format is the same as in [Routing Configuration](routing.md#ruleobject).
```

File: docs/en_US/config/dns.md
Comment (KevinZonda) [inline docs/en_US/config/dns.md]:
```suggestion
When this option is configured, V2Ray DNS will verify the returned IP and only return the addresses in the expectIPs list.
```

File: docs/en_US/config/dns.md
Comment (KevinZonda) [inline docs/en_US/config/dns.md]:
```suggestion
If this item is not configured, the origin IP address will be returned.
```

**Review by KevinZonda — CHANGES_REQUESTED** [review]
Thanks for you hard working! There are still some problems to solve before merging.

**Comment by rurirei** [comment]
honestly sorry, please use `squash and merge` to avoid not clear commit list.

File: docs/en_US/config/dns.md
Comment (rurirei) [reply docs/en_US/config/dns.md]: `ExpectIPs` is not a item for rewrite returned data, so i will insisted in my translation, but i made a supplement for here.

**Review by DuckSoft — APPROVED** [review]
So far, so good.

File: docs/en_US/config/dns.md
Comment (DuckSoft) [inline docs/en_US/config/dns.md:3]: Personally, I think this sentence need to be reorganised.
What do you think? @KevinZonda

**Comment by kslr** [comment]
Unfortunately, we need to close this PR
https://github.com/v2fly/v2fly-github-io/pull/75

**Pull request closed without merging**
